ably/realtime_channel.go

```diff
@@ -293,6 +294,13 @@ func (c *RealtimeChannel) Publish(name string, data interface{}) (Result, error)
 //
 // This implicitly attaches the channel if it's not already attached.
 func (c *RealtimeChannel) PublishAll(messages []*proto.Message) (Result, error) {
+	id := c.client.Auth.clientIDForCheck()
+	for _, v := range messages {
+		if v.ClientID != "" && id != wildcardClientID && v.ClientID != id {
+			// Spec RSL1g3,RSL1g4
+			return nil, fmt.Errorf("can't publish message with clientId %s the clientId is %s", v.ClientID, id)
```

Comment:
"Unable to publish message containing a clientId (%s) that is incompatible with the library clientId (%s)"